socket_xdp_linux_test.go

```diff
@@ -12,7 +13,9 @@ import (
 
 func TestSocketXDPGetInfo(t *testing.T) {
 	xdpsockfd, err := unix.Socket(unix.AF_XDP, unix.SOCK_RAW, 0)
-	if err != nil {
+	if errors.Is(err, unix.EPERM) {
```

Comment:
I understand avoiding extra indentation, but I feel it is more readable is we test the error type only if the error is not nil. I am sure `error.Is` test handles the _nil_ use case, but I do not want to have to check.
```
if err != nil {
    if errors.Is(err, unix.EPERM) {
        t.Skipf("creating AF_XDP socket not permitted")
    }
     t.Fatal(err)
}
```

Yes, this is only test code, but quite often test code is looked upon to find examples.